### ISSUE TITLE
NTP: serial GPS does not get initialized?

### ISSUE THREAD

**Issue by thatso** [opnsense/core#4144]
OPNsense 20.1.7-amd64

Not sure if I simply misunderstand the way things should work here ... I've recently connected a serial GPS receiver with UBX-M8030 chipset and consequently selected U-Blox as type on the page Services -> Network Time -> GPS, serial port cuau0 and baud rate 9600 (manufacturer's default for the connected receiver).
In GPS initialization -> Advanced I see as last line:
`$PUBX,41,1,0007,0003,4800,0*13`
I believe this should set the baud rate to 4800.
Then why does the log read:
`ntpd[50477]: GPS_NMEA(0) serial /dev/gps0 open at 9600 bps`

/dev/gps0 correctly links to /dev/cuau0:
```
# ls -l /dev/gps0
lrwxr-xr-x  1 root  wheel  10 May 29 05:00 /dev/gps0 -> /dev/cuau0
```
However, the baud rate remains at the initial 9600:
```
# stty < /dev/gps0
speed 9600 baud;
```
Meanwhile, `cat /dev/gps0` shows correct messages like $GNGGA and $GNRMC.

If I change the baud rate in the serial port settings from 9600 to 4800, the log file and port settings do indeed reflect this change:
`ntpd[39392]: GPS_NMEA(0) serial /dev/gps0 open at 4800 bps`
```
# stty < /dev/gps0
speed 4800 baud;
```
But at the same time, the status of the receiver changes from active peer to unavailable:
```
Status | Server | Ref ID | Stratum | Type | When | Poll | Reach | Delay | Offset | Jitter
Unreach/Pending | 127.127.20.0 | .GPS. | 0 | l | - | 16 | 0 | 0.000 | +0.000 | 0.000
```
and `cat /dev/gps0` no longer shows any output.

Now I wonder if the initialization sequence is not processed at all.

**Comment by thatso** [comment]
I've just found the closed bug #1712 from 3 years ago (not sure why it didn't show up before I've opened this one). It seems that the problem was never fixed: the initialization sequence is not sent.
The only difference is that my GPS has a real serial connection instead of USB.

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/core/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.
